// fetch-loader-react/dist/FetchData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; return newObj; } }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance"); }

function _iterableToArrayLimit(arr, i) { var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var FetchData = function FetchData(props) {
  var _useState = (0, _react.useState)(true),
      _useState2 = _slicedToArray(_useState, 2),
      loading = _useState2[0],
      setLoading = _useState2[1];

  var _useState3 = (0, _react.useState)(null),
      _useState4 = _slicedToArray(_useState3, 2),
      err = _useState4[0],
      setErr = _useState4[1];

  var _useState5 = (0, _react.useState)([]),
      _useState6 = _slicedToArray(_useState5, 2),
      res = _useState6[0],
      setRes = _useState6[1];

  (0, _react.useEffect)(function () {
    _asyncToGenerator(
    /*#__PURE__*/
    regeneratorRuntime.mark(function _callee() {
      var resFunc, _res;

      return regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              resFunc = {
                arrayBuffer: 'arrayBuffer',
                blob: 'blob',
                formData: 'formData',
                json: 'json',
                text: 'text'
              };
              _context.prev = 1;

              if (!(Array.isArray(props.fetch) && props.fetch[0])) {
                _context.next = 14;
                break;
              }

              _context.next = 5;
              return Promise.all(props.fetch);

            case 5:
              _res = _context.sent;

              if (!(props.type && resFunc[props.type])) {
                _context.next = 10;
                break;
              }

              _context.next = 9;
              return Promise.all(_res.map(function (res) {
                return res[resFunc[props.type]]();
              }));

            case 9:
              _res = _context.sent;

            case 10:
              setRes(_res);
              setLoading(false);
              _context.next = 16;
              break;

            case 14:
              if (!(!Array.isArray(props.fetch) || props.fetch.length === 0)) {
                _context.next = 16;
                break;
              }

              throw new Error("FetchData: fetch prop must be an array of json fetched promises");

            case 16:
              _context.next = 22;
              break;

            case 18:
              _context.prev = 18;
              _context.t0 = _context["catch"](1);
              setErr(_context.t0);
              setLoading(false);

            case 22:
            case "end":
              return _context.stop();
          }
        }
      }, _callee, null, [[1, 18]]);
    }))();
  }, [props]);
  return _react["default"].createElement(_react["default"].Fragment, null, props.children({
    loading: loading,
    err: err,
    res: res
  }));
};

var _default = FetchData;
exports["default"] = _default;
